java/core/src/main/java/co/worklytics/psoxy/gateway/impl/HealthCheckRequestHandler.java

```diff
@@ -51,53 +52,62 @@ private boolean isHealthCheckRequest(HttpEventRequest request) {
 
     private HttpEventResponse handle() {
         Set<String> missing =
-            sourceAuthStrategy.getRequiredConfigProperties().stream()
-                .filter(configProperty -> config.getConfigPropertyAsOptional(configProperty).isEmpty())
-                .map(ConfigService.ConfigProperty::name)
-                .collect(Collectors.toSet());
+                sourceAuthStrategy.getRequiredConfigProperties().stream()
+                        .filter(configProperty -> config.getConfigPropertyAsOptional(configProperty).isEmpty())
+                        .map(ConfigService.ConfigProperty::name)
+                        .collect(Collectors.toSet());
 
-        HealthCheckResult.HealthCheckResultBuilder healthCheckResult = HealthCheckResult.builder()
-            .configuredSource(config.getConfigPropertyAsOptional(ProxyConfigProperty.SOURCE).orElse(null))
-            .nonDefaultSalt(config.getConfigPropertyAsOptional(ProxyConfigProperty.PSOXY_SALT).isPresent())
-            .pseudonymImplementation(config.getConfigPropertyAsOptional(ProxyConfigProperty.PSEUDONYM_IMPLEMENTATION).orElse(null))
-            .missingConfigProperties(missing);
+        try {
+            Optional<String> targetHost = config.getConfigPropertyAsOptional(ProxyConfigProperty.TARGET_HOST);
 
+            if (targetHost.isEmpty() || StringUtils.isBlank(targetHost.get())) {
+                missing.add(ProxyConfigProperty.TARGET_HOST.name());
```

Comment:
Including TARGET_HOST as missing property; not a config property itself, but is something that might be configured. Also, adding it as a boolean flag in `HealthCheckResult` will force us to update the lib and put code to handle that validation in app side.

Reply:
I don't see why this try/catch is needed at all, how is adding a String to a Set going to fail??

Reply:
Not for that, just for reading the setting from config. It should not fail as it is an optional get, but just to be defensive here.